ArraySimulation/MPPT/GlobalMPPTAlgorithms/GlobalMPPTAlgorithm.py

```diff
@@ -0,0 +1,173 @@
+"""
+GlobalMPPTAlgorithm.py
+
+Author: Afnan Mir, Array Lead (2021).
+Contact: afnanmir@utexas.edu
+Created: 02/06/2021
+Last Modified: 02/08/2021
+
+Description: The GlobalMPPTAlgorithm class is a concrete base class that
+provides a common API for derived classes to use. The class enables users to
+calculate or predict voltage setpoints that would maximize the output power of
+the PVSource given a set of input conditions.
+"""
+# Library Imports.
+
+
+# Custom Imports.
+from ArraySimulation.MPPT.LocalMPPTAlgorithms.LocalMPPTAlgorithm import (
+    LocalMPPTAlgorithm,
+)
+from ArraySimulation.MPPT.LocalMPPTAlgorithms.PandO import PandO
+from ArraySimulation.MPPT.LocalMPPTAlgorithms.IC import IC
+from ArraySimulation.MPPT.LocalMPPTAlgorithms.FC import FC
+from ArraySimulation.MPPT.LocalMPPTAlgorithms.Ternary import Ternary
+from ArraySimulation.MPPT.LocalMPPTAlgorithms.Golden import Golden
+from ArraySimulation.MPPT.LocalMPPTAlgorithms.Bisection import Bisection
+
+
+class GlobalMPPTAlgorithm:
+    """
+    The GlobalMPPTAlgorithm class and derived classes attempt to solve P-V
+    curves with multiple local maxima and a single global maxima.
+    """
+
+    # The upper voltage bound that should be predicted by any model. We expect
+    # the PV to always be at open circuit voltage at this point. Adjustable
+    # based on the number of cells determined from the initialization.
+    MAX_VOLTAGE = 100
+
+    # The upper voltage bound for a single cell that should be predicted by any
+    # model. We expect the PV to always be at open circuit voltage at this
+    # point. The reference value was determined from experimentation from the
+    # Maxeon Gen III Bin Le1 solar cells, which have a rated voltage of .721V at
+    # standard conditions.
+    MAX_VOLTAGE_PER_CELL = 0.8
+
+    def __init__(
+        self,
+        numCells=1,
+        MPPTGlobalAlgoType="Default",
+        MPPTLocalAlgoType="Default",
+        strideType="Fixed",
+    ):
+        """
+        Sets up the initial source parameters.
+
+        Parameters
+        ----------
+        numCells: int
+            The number of cells that should be accounted for in the MPPT
+            algorithm.
+        MPPTGlobalAlgoType: String
+            The name of the global MPPT algorithm type.
+        MPPTLocalAlgoType: String
+            The name of the local MPPT algorithm type.
+        strideType: String
+            The name of the stride algorithm type.
+        TODO: Add stride argument to voltage sweep constructor.
+        """
+        self.MAX_VOLTAGE = self.MAX_VOLTAGE_PER_CELL * numCells
```

Comment:
As a class variable, this should be accessed as follows:
`GlobalMPPTAlgorithm.MAX_VOLTAGE = GlobalMPPTAlgorithm.MAX_VOLTAGE_PER_CELL * numCells`.
Also it kinda defeats the point to always delete the original value.
```
if GlobalMPPTAlgorithm.MAX_VOLTAGE_PER_CELL * numCells` < GlobalMPPTAlgorithm.MAX_VOLTAGE:
    GlobalMPPTAlgorithm.MAX_VOLTAGE = GlobalMPPTAlgorithm.MAX_VOLTAGE_PER_CELL * numCells
```

I suspect that this was my code.